chiselFrontend/src/main/scala/chisel3/core/Aggregate.scala

```diff
@@ -574,7 +591,10 @@ class Bundle(implicit compileOptions: CompileOptions) extends Record {
         clazz.getDeclaredField("$outer").get(this)  // doesn't work in all cases, namely anonymous inner Bundles
       } catch {
         case _: NoSuchFieldException =>
-          this.outerModule.getOrElse(reflectError(s"Unable to determine instance of outer class $outerClass"))
+          // If this Bundle has a ChildBinding, it's an anonymous inner Bundle of another Bundle
+          this.bindingOpt.collect { case ChildBinding(p) => p }
```

Comment:
Are there cases where this static priority system gets it wrong? For example, a Bundle that is directly a Module inner class but is also a child of another Bundle, so it would think the outer class is a Bundle instead of (properly) the Module? Code written that way might be a bit odd, but not that odd...

Reply:
That's a good point, I had waffled on whether to go static priority or not but it seems it probably shouldn't be. Looking into how to fix.